src/streaming/controllers/EventController.js

```diff
@@ -310,8 +313,10 @@ function EventController() {
     function _startEvent(eventId, event, events) {
         try {
             const currentVideoTime = playbackController.getTime();
+            const calculatedPresentationTimeInSeconds = event.calculatedPresentationTime / event.eventStream.timescale;
```

Comment:
We need to distinguish between application specific events and DASH specific events here. On receive mode is only required for application events, the DASH specific events should not be handled prior to their presentation time

Reply:
Suggestion: Add a new method _signalEvent which only dispatches application "onReceive" events. The handling for "onStart" events can remain the same